### ISSUE TITLE
ERR_UNSUPPORTED_DIR_IMPORT

### ISSUE THREAD

**Issue by rocerchen** [amannn/next-intl#378]
### Description

info  - Collecting page data .Error [ERR_UNSUPPORTED_DIR_IMPORT]: Directory import '/Users/chenrocer/Projects/myproject/node_modules/next-intl/dist/react-client' is not supported resolving ES modules imported from /Users/me/Projects/myproject/node_modules/next-intl/dist/index.mjs
    at new NodeError (node:internal/errors:399:5)
    at finalizeResolution (node:internal/modules/esm/resolve:319:17)
    at moduleResolve (node:internal/modules/esm/resolve:945:10)
    at defaultResolve (node:internal/modules/esm/resolve:1153:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:838:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:77:40)
    at link (node:internal/modules/esm/module_job:76:36) {
  code: 'ERR_UNSUPPORTED_DIR_IMPORT',
  url: 'file:///Users/me/Projects/myproject/node_modules/next-intl/dist/react-client'
}

### Mandatory reproduction URL (CodeSandbox or GitHub repository)

"next-intl": "^2.17.1"

### Reproduction description

when next build
info  - Collecting page data .Error [ERR_UNSUPPORTED_DIR_IMPORT]: Directory import '/Users/chenrocer/Projects/myproject/node_modules/next-intl/dist/react-client' is not supported resolving ES modules imported from /Users/me/Projects/myproject/node_modules/next-intl/dist/index.mjs
    at new NodeError (node:internal/errors:399:5)
    at finalizeResolution (node:internal/modules/esm/resolve:319:17)
    at moduleResolve (node:internal/modules/esm/resolve:945:10)
    at defaultResolve (node:internal/modules/esm/resolve:1153:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:838:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:77:40)
    at link (node:internal/modules/esm/module_job:76:36) {
  code: 'ERR_UNSUPPORTED_DIR_IMPORT',
  url: 'file:///Users/me/Projects/myproject/node_modules/next-intl/dist/react-client'
}

### Expected behaviour

next build success

**Comment by amannn** [comment]
Please provide a reproduction URL as mentioned in the issue template, as without it I can't help you unfortuantely.

**Comment by iffa** [comment]
@amannn Here you go: https://github.com/iffa/next-intl-378-reproduction/

```
➜  ~/Projects/next-intl-378-reproduction git:(main) npm run build

> next-intl-378-reproduction@0.1.0 build
> next build

- info Linting and checking validity of types  
- info Creating an optimized production build  
- info Compiled successfully
unhandledRejection Error [ERR_UNSUPPORTED_DIR_IMPORT]: Directory import '/Users/santeri/Projects/next-intl-378-reproduction/node_modules/next-intl/dist/react-client' is not supported resolving ES modules imported from /Users/santeri/Projects/next-intl-378-reproduction/node_modules/next-intl/dist/index.mjs
    at new NodeError (node:internal/errors:399:5)
    at finalizeResolution (node:internal/modules/esm/resolve:319:17)
    at moduleResolve (node:internal/modules/esm/resolve:945:10)
    at defaultResolve (node:internal/modules/esm/resolve:1153:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:838:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:77:40)
    at link (node:internal/modules/esm/module_job:76:36) {
  type: 'Error',
  code: 'ERR_UNSUPPORTED_DIR_IMPORT',
  url: 'file:///Users/santeri/Projects/next-intl-378-reproduction/node_modules/next-intl/dist/react-client'
}
- info Collecting page data .%                                                                                                                                   
```

**Comment by amannn** [comment]
@iffa Thanks, looking into it now!

**Comment by RickCole21** [comment]
downgrade to `v2.17.0` fix the issue, might be something wrong in v2.17.1.
